packages/runed/src/lib/utilities/ElementRect/ElementRect.svelte.ts

```diff
@@ -0,0 +1,104 @@
+import { extract } from "../extract/extract.js";
+import { useMutationObserver } from "../useMutationObserver/useMutationObserver.svelte.js";
+import { useResizeObserver } from "../useResizeObserver/useResizeObserver.svelte.js";
+import { watch } from "../watch/watch.svelte.js";
+import type { MaybeGetter } from "$lib/internal/types.js";
+
+type Rect = Omit<DOMRect, "toJSON">;
+
+export type ElementRectOptions = {
+	initialRect?: DOMRect;
+};
+
+/**
+ * Returns a reactive value holding the size of `node`.
+ *
+ * Accepts an `options` object with the following properties:
+ * - `initialSize`: The initial size of the element. Defaults to `{ width: 0, height: 0 }`.
+ * - `box`: The box model to use. Can be either `"content-box"` or `"border-box"`. Defaults to `"border-box"`.
+ *
+ * @returns an object with `width` and `height` properties.
+ *
+ * @see {@link https://runed.dev/docs/utilities/element-size}
+ */
+export class ElementRect {
+	#rect: Rect = $state({
```

Comment:
We can use `$state.frozen` here since `#rect` is reassigned entirely rather than mutated.